strings/isisogram.go

```diff
@@ -0,0 +1,69 @@
+// Checks if a given string is an isogram.
+// A first-order isogram is a word in which no letter of the alphabet occurs more than once.
+// A second-order isogram is a word in which each letter appears twice.
+// A third-order isogram is a word in which each letter appears three times.
+// wiki: https://en.wikipedia.org/wiki/Heterogram_(literature)#Isograms
+// Author: M3talM0nk3y
+
+package strings
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+)
+
+const (
+	firstOrder  int = 1
+	secondOrder int = 2
+	thirdOrder  int = 3
+)
+
+func hasDigit(text string) bool {
+	re := regexp.MustCompile(`\d`)
+	return re.MatchString(text)
+}
+
+func hasSymbol(text string) bool {
+	re := regexp.MustCompile(`[-!@#$%^&*()+]`)
+	return re.MatchString(text)
+}
+
+func IsIsogram(text string, order int) (bool, error) {
+	if order != firstOrder && order != secondOrder && order != thirdOrder {
```

Comment:
Check the [playground](https://go.dev/play/p/7dcoMZe0wZz) for a simpler implementation.

Reply:
Simplified the validation. Thanks for pointing that out.